R/harvest.R

```diff
@@ -0,0 +1,200 @@
+    dplyr::filter(no_harvest) |>
+    dplyr::group_by(watershed, origin, age, sim_year, return_year, return_sim_year) |>
+    dplyr::summarise(return_total = round(sum(return_total, na.rm = TRUE)),
+                     remaining_adults = round(sum(return_total, na.rm = TRUE) * .99, 0), # multipiles by .9 to remove 1 percent bycatch/hook mortality in trib (much lower than in ocean)
+                     actual_harvest = 0)
+  
+  trib_harvest_adults <- adults_after_ocean_harvest |>
+    dplyr::left_join(hab_capacity, by = "watershed") |>
+    dplyr::mutate(no_harvest = ifelse(sim_year %in% no_cohort_harvest_years, T, F), # Do we want to exclude hatchery here as well
+                  no_harvest = ifelse(restrict_harvest_to_hatchery_trib & origin == "natural", T, no_harvest)) |>
+    dplyr::filter(!no_harvest) |>
+    dplyr::rowwise() |>
+    dplyr::mutate(num_adults_required_after_harvest = ifelse(intelligent_crr_harvest,
+                                                             round(spawner_df[watershed, sim_year] *
+                                                                     return_prop[origin, return_year] * crr_scaling, 0), 0), # this is capturing total
+                  age = return_sim_year - sim_year,
+                  trib_harvest = tributary_harvest_percentage[watershed],
+                  adults_after_trib_harvest = round((return_total * (1 - trib_harvest)), 0),
+                  combined_harvest_number = (return_total - adults_after_ocean_harvest) + (return_total - adults_after_trib_harvest),
+                  adults_after_harvest = return_total - combined_harvest_number,
+                  utalize_crr_totals = ifelse(num_adults_required_after_harvest > adults_after_harvest & intelligent_crr_harvest, TRUE, FALSE),
+                  utalize_hab_totals = ifelse(habitat_capacity > adults_after_harvest & intelligent_habitat_harvest, TRUE, FALSE),
+                  remaining_adults = round(dplyr::case_when(utalize_crr_totals ~ min(num_adults_required_after_harvest, return_total),
+                                                            utalize_hab_totals ~ min(habitat_capacity, return_total),
+                                                            T ~ max(adults_after_harvest, 0))),
+                  actual_harvest = round(dplyr::case_when(utalize_crr_totals ~ max(return_total - num_adults_required_after_harvest, 0),
+                                                          utalize_hab_totals ~ max(return_total - habitat_capacity, 0),
+                                                          T ~ min(combined_harvest_number, return_total)), 0)) |>
+    dplyr::select(watershed, origin, age, remaining_adults, actual_harvest)
+  
+  
+  adults_after_harvest <- dplyr::bind_rows(trib_no_harvest_adults,
+                                           trib_harvest_adults)
+  # Set to always allow - apply to no harvest and harvest adults, this will break crr logic (TODO talk with rene about that)
```

Comment:
lets also go through and address these todos/remove if they are no longer relevant (in fall)